### DISCUSSION TITLE
Encoding large integers as uint64 in MessagePack

### DISCUSSION THREAD

**Discussion by jackkru69** [kriszyp/msgpackr#143] [General]
Hey everyone,

I'm currently working with MessagePack, and I need to encode large integers as uint64. However, I don't expect to deal with floats in my use case—I'm specifically working with integers only.

For instance, I tried packing the number 1000000000000000000 as a sum, and from what I understand, the result should be something like CF0DE0B6B3A7640000.

Could someone confirm whether the CF type in MessagePack corresponds to uint64, or if there's something I'm missing here?

Thanks in advance!

**Comment by kriszyp** [comment]
Have you tried 1000000000000000000n? This is the JavaScript literal for a 64-bit (or more) integer. (1000000000000000000 is a float in JS.)

**Reply by jackkru69** [reply]
ok, I'll try, but can I encode, for example, 1723804081783 not as a float?

**Reply by kriszyp** [reply]
You can encode any number as a 64-bit integer, by using BigInt, as long as it fits in 64 bits.

**Comment by matthaeis** [comment]
Hello @kriszyp and @jackkru69,

i have a similiar issue. uint 64-bit integer shall be mapped to 0xCF .. .. .. .. .. .. .. .., but actually they are mapped to 0xD3 (if they are small enough)

E.g.: bigint 1695129253235n is mapped to D30000018AAD93B573 but it could also be CF0000018AAD93B573

My expectation would be: When it is positive, use uint64 (0xCF), only when it is negative use int64 (0xd3)

For my understanding the fix can be done in line 526 of pack.js by switching the two if conditions:
First check if it fits into unsigend64-bit (0xCF and only if not move on with 0xD3)

Do you agree?
